### PR TITLE
Speed up daml-sdk-head Maven installation

### PR DESCRIPTION
**Pull request by cocreature**
This PR speeds up the Maven installation by doing 4 things:

1. Disable TS builds. This doesn’t make sense since we never install
them.
2. Add a --skip-sdk option that skips the SDK installation.
3. Add a --skip-jar-docs option that skips javadoc and more
importantly scaladoc.
4. Add a --scala-version=$VERSION option to only build against the
given version.

Unfortunately there are no tests and the CLI is a bit messy (you need
--scala-version=$VERSION and not --scala-version $VERSION).

Both are not great but this is a development only tool and it is bash
so my motivation for fixing it is rather low.

Given that this is growing in complexity a fair bit, I think we might
be better off as a long-term solution to move this towards at least
python or maybe even Haskell or Scala. But definitely nothing for this
PR.

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: release/src/Options.hs [hunk 1]
```diff
@@ -18,10 +20,20 @@ data Options = Options
...
+  , optIncludeDocs :: IncludeDocs
   } deriving (Eq, Show)
 
 optsParser :: Parser Options
 optsParser = Options
   <$> (PerformUpload <$> switch (long "upload" <> help "upload java/scala artifacts to Maven Central and typescript artifacts to the npm registry."))
   <*> option str (long "release-dir" <> help "specify full path to release directory")
   <*> switch (long "install-head-jars" <> help "install jars to ~/.m2")
+  <*> fmap (IncludeTypescript . not) (switch (long "no-ts" <> help "Do not build and upload typescript packages"))
+  <*> fmap (fromMaybe defaultScalaVersions) (optional (some (strOption (long "scala-version" <> help "Scala version to build JARs for"))))
```
Comment (aherrmann-da) [inline release/src/Options.hs:34]: Perhaps easier with `some ... <|> pure defaultScalaVersions`?

**Review by aherrmann-da — APPROVED** [review]
Nice! How much faster does it get?

File: release/src/Options.hs (hunk 1 above)
Comment (cocreature) [reply release/src/Options.hs:34]: good idea, done

**Comment by cocreature** [comment]
> Nice! How much faster does it get?

My testcase is modifying `com.daml.lf.data.ImmArray` after first building everything:

With this change and `daml-sdk-head --skip-sdk --skip-jar-docs --scala-version=2.13.5` (note that this is still slightly worse than it could be, the default Scala version is better):
~76s

Without this change and regular `daml-sdk-head`:
~18.45min (note the unit)

Not entirely sure I got the same number of cache hits but it definitely does match up with the numbers I would have guessed based on experience.

**Pull request merged**